testing/common_testing_util.py

```diff
@@ -107,3 +121,96 @@ def printv(string, verbose=1, type='normal'):
+        config.json to construct the BidsDataFrame
+    """
+
+    # Create the bids frame.
+    bids_df = BidsDataframe(loader_parameters,
+                            str(path_data_multi_sessions_contrasts_tmp),
+                            derivatives=True)
+
+    # Assert new dataframe EQUALs an empty pandas dataframe
+    assert bids_df.df.equals(pd.DataFrame())
+
+
+def bids_dataframe_comparison_framework(loader_parameters: dict, target_csv: str):
+    """Main test function used to set up a CSV comparison between expected vs the output from the test
+    Args:
+        loader_parameters (dict): a dictionary object containing all loader parameters necessary derived from
+        config.json to construct the BidsDataFrame
+
+        target_csv (str): the filename string  of a CSV which indicate the goal of the unit test scenario. It is
+        derived from the bids_dataframe after stripping out its PATH column.
+    """
+    # Create the bids frame.
+    bids_df = BidsDataframe(loader_parameters,
+                            str(path_data_multi_sessions_contrasts_tmp),
+                            derivatives=True)
+
+    # Drop path as that can varies across runs.
+    df_test = bids_df.df.drop(columns=[BidsDataFrameKW.PATH])
+
+    # Sorting to ensure consistencies when comparing with ground truth.
+    df_test = df_test.sort_values(by=[BidsDataFrameKW.FILENAME]).reset_index(drop=True)
+
+    # Generate full path to the target reference that we hope to match (e.g. df_ref.csv or any other scenario specific
+    # CSV file name
+    csv_ref = os.path.join(loader_parameters[LoaderParamsKW.PATH_DATA][0], target_csv)
+
+    # Generate full path for the csv which is produced from Bids Dataframe so that we can save the CSV to that location
+    csv_test = os.path.join(loader_parameters[LoaderParamsKW.PATH_DATA][0], "df_test.csv")
+    df_test.to_csv(csv_test, index=False)
+
+    # Calculate differences between the generated versus the ground truth by comparing the output csv (csv_test) with
+    # the target reference CSV from the data rep (csv_ref)
+    diff = csv_diff.compare(csv_diff.load_csv(open(csv_ref)), csv_diff.load_csv(open(csv_test)))
+
+    # Any modification is in relation to csv_test - csv_ref direction. i.e. Add = what did the csv_diff Add in
+    # relation to the csv_ref.
+    assert diff == {'added': [], 'removed': [], 'changed': [], 'columns_added': [], 'columns_removed': []}
+
+
+def get_multi_default_case() -> dict:
+    """Generate a default case loader json for the multi-session configuration JSON by:
+    1) load the default config.json
+    2) load its loader parameters
+    3) overwrite its various information in relation to the data_multi unit test data most notably for
+        a) target sessions
+        b) path data
+        c) target suffix
+        d) contrast parameters
+
+    returns:  the unit test ready loader parameters
+    """
```

Comment:
Similar to above, this one function provides a baseline loader parameter dictionary to work for all multi-session unit tests scenarios (to be overwritten in specific scenario). We will eventually adopt a similar approach to generate a common loader test scenarios.